suse_migration_services/units/prepare.py

```diff
@@ -101,6 +101,28 @@ def main():
     cloud_register_metadata = os.sep.join(
         [root_path, 'var', 'lib', 'cloudregister']
     )
+    zypper_log_path = os.sep.join(
+        [root_path, 'var', 'log', 'zypper.log']
+    )
+    if os.path.exists(zypper_log_path):
+        if not os.path.exists('/var/log'):
+            log.info('Creating /var/log')
+            Path.create('/var/log')
```

Comment:
You don't have to check for the presence of /var/log, that will always exist because of the FHS standard. However you have to check on /var/log/zypper.log and create an empty file if it does not exist. The reason is you can't bind mount root_path/var/log/zypper.log to /var/log/zypper.log if this target does not exist. This is the same concept as if you would bind mount a directory. The target must exist

Reply:
Also I think there are too many occurrences of var/log/zypper.log

Reply:
```python
zypper_log_file = os.sep.join(
    [root_path, 'var', 'log', 'zypper.log']
)
if os.path.exists(zypper_log_file):
    zypper_host_log_file = zypper_log_file.replace(root_path, os.sep)
    if not os.path.exists(zypper_host_log_file):
        with open(zypper_host_log_file, 'w'):
            # we bind mount the system zypper log file but the mount target
            # does not exist. Create it as empty file prior bind mounting
            pass
    
    ...rest of the code
```

Reply:
Done. 
`replace(root_path, os.sep)` would not work (it would be `//path`) so I changed it.